src/Libraries/Microsoft.Extensions.AI/Utilities/AIJsonUtilities.Schema.cs

```diff
@@ -278,24 +276,24 @@ JsonNode TransformSchemaNode(JsonSchemaExporterContext ctx, JsonNode schema)
                 {
                     objSchema.Add(AdditionalPropertiesPropertyName, (JsonNode)false);
                 }
-            }
-
-            if (ctx.Path.IsEmpty)
-            {
-                // We are at the root-level schema node, update/append parameter-specific metadata
 
                 // Some consumers of the JSON schema, including Ollama as of v0.3.13, don't understand
                 // schemas with "type": [...], and only understand "type" being a single value.
                 // STJ represents .NET integer types as ["string", "integer"], which will then lead to an error.
-                if (TypeIsArrayContainingInteger(schema))
+                if (TypeIsArrayContainingInteger(objSchema))
```

Comment:
@SteveSandersonMS I think we discussed this in the past, but I'm somewhat confused by this rule. If Ollama doesn't support arrays in type keywords, shouldn't we always be removing them? 

Note that `"integer"` doesn't necessarily get bundled with `"string"`. For example, the schema for `int?` would have a type keyword of `["integer", "null"]` which this transformation would render as `"integer"`.